odoo/addons/base/ir/ir_ui_view.py

```diff
@@ -237,6 +238,12 @@ def _compute_model_data_id(self):
             view = self.browse(data['res_id'])
             view.model_data_id = data['id']
 
+    def _search_model_data_id(self, operator, value):
+        name = 'name' if isinstance(value, basestring) else 'id'
+        domain = [('model', '=', 'ir.ui.view'), (name, operator, value)]
+        datas = self.env['ir.model.data'].search(domain)
+        return ('id', 'in', datas.mapped('res_id'))
```

Comment:
Oops, that's a mistake indeed!
Search performance should be roughly similar, I guess, but I haven't tested it.